clang/lib/Frontend/InitPreprocessor.cpp

```diff
@@ -1006,6 +1006,8 @@ static void InitializePredefinedMacros(const TargetInfo &TI,
   else if (LangOpts.hasDWARFExceptions() &&
            (TI.getTriple().isThumb() || TI.getTriple().isARM()))
     Builder.defineMacro("__ARM_DWARF_EH__");
+  else if (LangOpts.hasWasmExceptions() && TI.getTriple().isWasm())
+    Builder.defineMacro("__USING_WASM_EXCEPTIONS__");
```

Comment:
How about just calling this `__WASM_EXCEPTIONS__`?

Reply:
I like it more too. I started using `__USING_WASM_EXCEPTIONS__` mostly because to be consistent with `__USING_SJLJ_EXCEPTIONS__`, which Wasm EH share many traits. There are many lines in libc++abi and libunwind where both SjLj and Wasm include or exclude by `ifdef`s like this
https://github.com/llvm/llvm-project/blob/bdfb04a63d73c31ee75395064762d0d6ccb45819/libcxxabi/src/cxa_personality.cpp#L645

Not sure if it was really necessary to be consistent with it though. But anyway to do that we need to fix it everywhere, both in LLVM and Emscripten and libraries (libc++abi and libunwind). And it is likely other toolchains (like WASI) users have been defining `__USING_WASM_EXCEPTIONS__` somewhere in their build system already given that we have not been defining this in Clang so far. Do you think it's worth changing or feasible?

Reply:
I see the `__USING_` prefix does predate out work on Wasm exceptions so there is some precedent.  

I see __USING_SJLJ_EXCEPTIONS__ dating all the way back to 2009: 2f5c75e17dc0b2df841db396de22c35b95cb7f41.   However, I don't see any other predefined macros using the `__USING_` prefix, so it might be worth removing it now before it gets even harder to remove.

Presumably it would be quite easy for downstream toolchains to switch to using `__WASM_EXCEPTIONS__` either before or after this change lands.. and then remove the manual define at some point in the future.

Reply:
Changing it in LLVM and Emscripten is not a big problem, but I'm still somewhat worried about non-Emscripten toolchain users who have `-D__USING_WASM_EXCEPTIONS__` somewhere in their build configuration. I can add some comment on the previous issues for Wasm EH library porting and change the parts of WASI toolchain that uses the macro.

And why do you think we should remove this define here in Clang (assuming we change this to whatever we like)? This is consistent with what other exception modes are doing, and people need this `define` somewhere.

Reply:
I don't think we should remove the define here, I just want us to consider carefully the name of the define if we are going to have the compiler generate it.

The removal I was referring to was the `-D...` flags used in emscripten and wasi-sdk, which will be redundant once this change land (and they convert to using the new macro).

Reply:
I see. And come to think of it, as you said, I may not need to worry about the predefined macros in other users because they might be redundant but this will not break them because we define the macro in Clang.

Will update this to `__WASM_EXCEPTIONS__`, and update the macro in other parts of LLVM and Emscripten as follow-ups.